main/src/ca/uwaterloo/flix/language/phase/typer/PrimitiveEffects.Methods.json

```diff
@@ -1,11 +1,20 @@
 {
   "methods": {
     "java.lang.Boolean::getBoolean": "Env, IO",
-
+    "java.lang.Runtime::getRuntime": "IO",
+    "java.lang.Runtime::exit": "Sys, IO",
+    "java.lang.Runtime::addShutdownHook": "Exec, Sys, IO",
+    "java.lang.Runtime::removeShutdownHook": "Sys, IO",
+    "java.lang.Runtime::halt": "Sys, IO",
+    "java.lang.Runtime::exec": "Env, Exec, IO",
     "java.lang.Runtime::availableProcessors": "Env, IO",
-    "java.lang.Runtime::freeMemory": "Env, IO",
-    "java.lang.Runtime::totalMemory": "Env, IO",
+    "java.lang.Runtime::freeMemory": "Env, NonDet, IO",
+    "java.lang.Runtime::totalMemory": "Env, NonDet, IO",
     "java.lang.Runtime::maxMemory": "Env, IO",
+    "java.lang.Runtime::gc": "Sys, IO",
```

Comment:
I am not sure this is Sys... In the sense that doing GC is kind of almost invisible.

Reply:
Yeah I have the same consideration. But at the same time it does interact with the JVM (but in a completely safe manner (except maybe you could potentially pause execution if you allocate a bunch of objects and then run `gc`?))